src/domain/parties/getPartiesByTypeAndID.js

```diff
@@ -97,7 +98,7 @@ const getPartiesByTypeAndID = async (headers, params, method, query, span, cache
     if (destination) {
       const destParticipantModel = await participant.validateParticipant(destination)
       if (!destParticipantModel) {
-        const proxyId = await proxyCache.lookupProxyByDfspId(destination)
+        const proxyId = proxyEnabled && await proxyCache.lookupProxyByDfspId(destination)
```

Comment:
Didn't we discuss to implement the logic for disabling proxyCache inside proxyCache?
 So if `proxyCacheConfig.enabled !== false`, any methods will return `null` or `false`

Reply:
More like saying passing a flag into a database client to make it optional. That is not the responsibility of the client library.

Reply:
Having the proxy cache in the execution path when its disabled and using its return values to decide a behavior we already know ahead of time seems worse to me.